dataclients/kubernetes/clusterclient_test.go

```diff
@@ -266,7 +266,7 @@ func TestLoggingInterval(t *testing.T) {
 	defer log.SetOutput(os.Stderr)
 
 	countMessages := func() int {
-		return strings.Count(out.String(), "Error transforming external hosts")
+		return strings.Count(out.String(), "Ignoring route 0: service not found: default/myapp")
```

Comment:
I think we need to refactor this counting thing

Reply:
Why? It uses route index. We can do i+1 but I think this will be more confusing. Or do you have another idea?

Reply:
@AlexanderYastrebov do we break if we have endpoints but no service?
If so: Why? 

Test case looks like this and I would expect it to work (likely it breaks because of service port lookup, which might be the right thing):
```
apiVersion: zalando.org/v1
kind: RouteGroup
metadata:
  name: myapp
spec:
  hosts:
  - example.org
  backends:
  - name: myapp
    type: service
    serviceName: myapp
    servicePort: 80
  defaultBackends:
  - backendName: myapp
  routes:
  - path: /app
---
apiVersion: v1
kind: Endpoints
metadata:
  name: myapp
subsets:
- addresses:
  - ip: 10.2.4.8
  - ip: 10.2.4.16
  ports:
  - port: 80
```

Reply:
I was talking about the whole test function, I remember I had to change the manifest before for validation, the more we tighten validation the more the need to change in this test case https://github.com/zalando/skipper/commit/92e0d11af311949faf68d55635ec302c66d38b1c

Reply:
> do we break if we have endpoints but no service?
> If so: Why?

The code checks service existence and it uses it to resolve port, see https://github.com/zalando/skipper/blob/ab3a18d1d4441585d069621a9adadf418fcc2cf0/dataclients/kubernetes/routegroup.go#L171-L193

Reply:
The scope of this change is to fix the bug we observe. I think we should handle test improvement or reconsider endpoint port handling in a separate effort (if at all).